client/my-sites/themes/theme-activation-confirmation-modal.jsx

```diff
@@ -0,0 +1,274 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { translate } from 'i18n-calypso';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import { Dialog } from '@automattic/components';
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import ExternalLink from 'calypso/components/external-link';
+import TrackComponentView from 'calypso/lib/analytics/track-component-view';
+import FormLabel from 'calypso/components/forms/form-label';
+import FormRadio from 'calypso/components/forms/form-radio';
+import {
+	getActiveTheme,
+	getCanonicalTheme,
+	hasActivatedTheme,
+	themeHasAutoLoadingHomepage,
+	isActivatingTheme,
+	isUsingRetiredTheme,
+	isThemeActive,
+	getPreActivateThemeId,
+} from 'calypso/state/themes/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import {
+	acceptActivateModalWarning,
+	hideActivateModalWarning,
+	activate as activateTheme,
+} from 'calypso/state/themes/actions';
+
+/**
+ * Style dependencies
+ */
+import './theme-activation-confirmation-modal.scss';
+
+class ThemeActivationConfirmationModal extends Component {
+	static propTypes = {
+		source: PropTypes.oneOf( [ 'details', 'list', 'upload' ] ).isRequired,
+		theme: PropTypes.shape( {
+			author: PropTypes.string,
+			author_uri: PropTypes.string,
+			id: PropTypes.string,
+			name: PropTypes.string,
+		} ),
+		hasActivated: PropTypes.bool.isRequired,
+		isActivating: PropTypes.bool.isRequired,
+		hasAutoLoadingHomepage: PropTypes.bool,
+		siteId: PropTypes.number,
+		isCurrentThemeRetired: PropTypes.bool,
+		onClose: PropTypes.func,
+		installingThemeId: PropTypes.string,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			homepageAction: 'keep_current_homepage',
+		};
+	}
+
+	handleHomepageAction = ( event ) => {
+		this.setState( { homepageAction: event.currentTarget.value } );
+	};
+
+	closeModalHandler = ( action = 'dismiss' ) => () => {
+		const {
+			hasAutoLoadingHomepage,
+			installingThemeId,
+			isCurrentThemeRetired,
+			siteId,
+			source,
+		} = this.props;
+		const isSolelyRetiredThemeModal = isCurrentThemeRetired && ! hasAutoLoadingHomepage;
+		const tracksPrefix = isSolelyRetiredThemeModal ? 'retired_theme' : 'autoloading_homepage';
+
+		if ( 'activeTheme' === action ) {
+			this.props.acceptActivateModalWarning( installingThemeId );
+			const keepCurrentHomepage = this.state.homepageAction === 'keep_current_homepage';
+			recordTracksEvent( `calypso_theme_${ tracksPrefix }_modal_activate_click`, {
```

Comment:
Hm, this will be problematic. We don't want to rename events that are still valid, otherwise we lose all history and usefulness of that event. We also don't want to use string interpolation in event names; it makes them very difficult to find in the codebase.

We can record multiple Tracks events at the same time – can we leave the existing ones as they are, then add another `recordTracksEvent` call specifically for the retired stuff? It's fine that the button language or presentation may have needed to change, as long as the action taken matches with the event fired.

Reply:
Sure thing, let me know if the new ones are okay!